File: CPP/ByRound/1545/1545_A_AquaMoon_and_Strange_Sort.cpp
// Problem: CF 1545 A - AquaMoon and Strange Sort
// https://codeforces.com/contest/1545/problem/A

/*
A. AquaMoon and Strange Sort
Algorithms/Techniques: Sorting, Greedy, Array Manipulation

Time Complexity: O(n log n) per test case due to sorting
Space Complexity: O(n) for storing the array

The problem involves sorting an array using adjacent swaps with direction flips,
with the constraint that all friends must face right at the end.
The approach separates elements into two groups based on their positions (even/odd indices),
sorts each group independently, and checks if placing elements in order maintains non-decreasing property
while ensuring final directions are right.
*/

#include <stdio.h>
#include <algorithm>

using namespace std;
const int N = 1e5 + 7;
int T_T, n, i, j, a[2][N]; // a[0] stores even-indexed elements, a[1] stores odd-indexed elements

int main() {
  for (scanf("%d", &T_T); T_T--;) {
    for (scanf("%d", &n), i = 1; i <= n; ++i)
      scanf("%d", a[i & 1] + (i + 1) / 2); // Store elements in even/odd indexed arrays
    sort(a[0] + 1, a[0] + n / 2 + 1); // Sort the even-indexed group
    sort(a[1] + 1, a[1] + (n + 1) / 2 + 1); // Sort the odd-indexed group
    for (j = 0, i = 1; i <= n; ++i)
      if (a[i & 1][(i + 1) / 2] < a[(i - 1) & 1][i / 2]) { // Check if current element is less than previous one when properly ordered
        puts("NO");
        j = 1;
        break;
      }
    if (!j)
      puts("YES");
  }
}


// https://github.com/VaHiX/codeForces/

File: Python/ByRound/1545/1545_A_AquaMoon_and_Strange_Sort.py
# Problem: CF 1545 A - AquaMoon and Strange Sort
# https://codeforces.com/contest/1545/problem/A

"""
AquaMoon and Strange Sort

Algorithm/Techniques: Sorting, parity check

Time Complexity: O(n log n) due to sorting operations
Space Complexity: O(n) for storing the array and its sorted version

This problem checks if we can sort an array by swapping adjacent elements such that 
the final arrangement is non-decreasing and all elements end up facing right.
The key insight is that we can only swap adjacent elements, so we're essentially checking 
if it's possible to sort the array using adjacent swaps. Since we can perform any number of 
adjacent swaps, the array can always be sorted using bubble sort or similar methods.

However, in this specific case, since we are told that direction is flipped on each swap,
the parity of swaps affects directions. For the final state to have all friends facing right, 
an even number of flips must occur for each element.
This solution uses an observation: if both odd and even indexed elements (when considering 0-based indexing) 
are sorted in their respective positions when comparing with original array, then sorting is possible.
"""

t = int(input())
for _ in range(t):
    n = int(input())
    a = list(map(int, input().split()))
    b = sorted(a)
    ans = "NO"
    # Check if elements at even indices (0, 2, 4, ...) are the same when sorted
    if sorted(a[::2]) == b[::2]:
        ans = "YES"
    print(ans)


# https://github.com/VaHiX/codeForces/